tests/dashboard/test_collection.py

```diff
@@ -114,3 +114,11 @@ def test_collection_is_published_toggle_view(db, admin_client, collection):
     admin_client.post(url)
     collection.refresh_from_db()
     assert collection.is_published
+
+
+def test_collection_add_not_published(admin_client, collection, site_settings):
+    collection.is_published = False
+    collection.save()
+    data = {'homepage_collection': collection.pk}
+    response = admin_client.post(reverse('dashboard:collection-list'), data)
+    assert response.status_code == 200
```

Comment:
I recommend to test if we really didn't add the collection to `homepage_collection` field in `SiteSettings`. http 200 doesn't tell anything when you read such test.